modules/alias/main.tf

```diff
@@ -167,8 +167,12 @@ resource "aws_lambda_event_source_mapping" "this" {
     for_each = try(each.value.filter_criteria, null) != null ? [true] : []
 
     content {
-      filter {
-        pattern = try(each.value["filter_criteria"].pattern, null)
+      dynamic "filter" {
+        for_each = try(flatten([each.value.filter_criteria]), [])
+
+        content {
+          pattern = try(filter.value.pattern, null)
+        }
```

Comment:
For reference: https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/v7.6.0/main.tf#L379-L391